presto-main/src/test/java/com/facebook/presto/sql/query/TestSubqueries.java

```diff
@@ -181,6 +181,14 @@ public void testCorrelatedLateralWithGroupBy()
                 UNSUPPORTED_CORRELATED_SUBQUERY_ERROR_MSG);
     }
 
+    @Test
+    public void testLateralWithUnnest()
+    {
+        assertions.assertFails(
+                "SELECT * FROM (VALUES ARRAY[1]) t(x), LATERAL (SELECT * FROM UNNEST(x))",
+                UNSUPPORTED_CORRELATED_SUBQUERY_ERROR_MSG);
```

Comment:
Can there be some _positive_ test for this?

Reply:
I am not sure... Maybe after changes @sopel39 did recently for correlated subqueries we support some subqueries with correlation in values.

Reply:
`ValuesNode` can be converted to `ProjectNode` with correlated symbol assignment. If such symbol is not later used with symbols from another table in subquery, then we could just unwind such query, which would be equivalent to:
```
SELECT * FROM (VALUES ARRAY[1]) t(x) CROSS JOIN UNNEST(x)",
```

Reply:
The query you posted is typical CROSS JOIN UNNEST which has nothing with correlated subqueries (but the fact it can use symbols from left side of join like LATERAL).

Reply:
> ValuesNode can be converted to ProjectNode

Unless Values have more rows.